common/persistence/client/fx.go

```diff
@@ -152,3 +167,77 @@ func HealthSignalAggregatorProvider(
 
 	return persistence.NoopHealthSignalAggregator
 }
+
+func ClusterMetadataManagerProvider(factory Factory) (persistence.ClusterMetadataManager, error) {
+	return factory.NewClusterMetadataManager()
+}
+func ClusterMetadataManagerLifetimeHooks(
+	lc fx.Lifecycle,
+	manager persistence.ClusterMetadataManager,
+) {
```

Comment:
these would look a little nicer on one line (rename `manager` to `m` if you want to stay ≤ 100 columns)